Examples/Java/Sources/Board.java

```diff
@@ -0,0 +1,88 @@
+//
+// Board.java
+// Autogenerated by plank
+//
+// DO NOT EDIT - EDITS WILL BE OVERWRITTEN
+// @generated
+//
+
+package com.pinterest.models;
+
+import android.support.annotation.IntDef;
+import com.google.auto.value.AutoValue;
+import java.net.URI;
+import java.util.Set;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringDef;
+import java.lang.annotation.Retention;
+import java.util.Date;
+import java.util.List;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Map;
+
+public interface BoardModel {
+    @Nullable String name();
+    Image image();
+    @Nullable Map<String, Integer> counts();
+    @Nullable Date createdAt();
+    @Nullable Set<User> contributors();
+    @Nullable String descriptionText();
+    @Nullable Map<String, String> creator();
+    @Nullable URI url();
+}
+
+public interface BoardModelBuilder {
+    Builder setName(@Nullable String value);
+    Builder setImage(Image value);
+    Builder setCounts(@Nullable Map<String, Integer> value);
+    Builder setCreatedAt(@Nullable Date value);
+    Builder setContributors(@Nullable Set<User> value);
+    Builder setDescriptionText(@Nullable String value);
+    Builder setCreator(@Nullable Map<String, String> value);
+    Builder setUrl(@Nullable URI value);
+    Builder setName(String value);
+    Builder setCounts(Map<String, Integer> value);
+    Builder setCreatedAt(Date value);
+    Builder setContributors(Set<User> value);
+    Builder setDescriptionText(String value);
+    Builder setCreator(Map<String, String> value);
+    Builder setUrl(URI value);
+}
+
+@AutoValue
+public abstract class Board implements ModelModel {
+
+    public abstract @Nullable String name();
+    public abstract Image image();
+    public abstract @Nullable Map<String, Integer> counts();
+    public abstract @Nullable Date createdAt();
+    public abstract @Nullable Set<User> contributors();
+    public abstract @Nullable String descriptionText();
+    public abstract @Nullable Map<String, String> creator();
+    public abstract @Nullable URI url();
+    public static Builder builder() {
+        return new AutoValue_Board.Builder();
+    }
+    abstract Builder toBuilder();
+    @AutoValue.Builder
+    public abstract static class Builder implements ModelModelBuilder {
+    
+        public abstract Builder setName(@Nullable String value);
+        public abstract Builder setImage(Image value);
+        public abstract Builder setCounts(@Nullable Map<String, Integer> value);
+        public abstract Builder setCreatedAt(@Nullable Date value);
+        public abstract Builder setContributors(@Nullable Set<User> value);
+        public abstract Builder setDescriptionText(@Nullable String value);
+        public abstract Builder setCreator(@Nullable Map<String, String> value);
+        public abstract Builder setUrl(@Nullable URI value);
+        public abstract Builder setName(String value);
```

Comment:
btw, having both the nullable and the non-null type parameters is redundant.
ideas:

1. `clearName()` and `setName(String)`
2. `setName(@Nullable String)`

for this case, 2 seems better.